backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/VirtualQueueRepository.java

```diff
@@ -33,4 +34,6 @@
     public List<Token> getTokens(UUID shopId);
 
     public GetShopsByShopOwnerResponse getShopsByShopOwner(String shopOwnerId);
+
+	public DeleteTokenResponse deleteToken(UUID tokenId, Boolean isLoggedin);
```

Comment:
Instead of isLoggedIn can pass shopOwnerUid and make sure that the status of the token is being changed by the shopOwner of the shop to which the token belongs??Just a thought.

Reply:
That's a nice idea. But as far as I understand, according to our logic, no other logged in user has the option to delete these particular tokens except the shop-owner of that particular shop itself. So do we need to check this here? Just wondering, I might be wrong also here.

Reply:
yes true but clients can change the source code and send the request maybe? For better security advised to consider shopOwnerUid.

Reply:
Oh, I'm not aware. Are clients allowed to do that kind of stuff? If so, this check can be implemented.

Reply:
Will bring this up with hosts and can take a call as per their suggestion

Reply:
Is it possible to change the source code ?? Like they can change the cookies but the source code is uploaded on a server right how will they change that? Just wondering.

Reply:
Yup I said the same Ishita. Let's discuss this with Bavanya later.

Reply:
yeah cool and we can discuss it with hosts as well to get better clarity on what is the best implementation here.